ui/plugins/pinboard.js

```diff
@@ -249,10 +251,16 @@ treeherder.controller('PinboardCtrl', [
                 if (!$scope.newEnteredBugNumber) {
                     $scope.toggleEnterBugNumber(false);
                 } else {
-                    $log.debug("new bug number to be saved: ",
-                               $scope.newEnteredBugNumber);
-                    thPinboard.addBug({id:$scope.newEnteredBugNumber});
-                    $scope.toggleEnterBugNumber(false);
+                    var num = parseInt($scope.newEnteredBugNumber);
+                    if (num == $scope.newEnteredBugNumber && num > 0) {
+                        $log.debug("new bug number to be saved: ",
+                                   $scope.newEnteredBugNumber);
+                        thPinboard.addBug({id:$scope.newEnteredBugNumber});
+                        $scope.toggleEnterBugNumber(false);
+                        return true;
+                    } else {
+                        return false;
+                    }
```

Comment:
Parse the value in the input to be an integer. If that new integer is still the same string representation AND it's above 0, let it be saved. (This basically matches the pattern being checked https://github.com/mozilla/treeherder/blob/30d3ca1b4882d61f19a237b2fd66796d8c597ed1/ui/js/directives/treeherder/main.js#L151 )
Return true in the success case and false in the failure case so we can know elsewhere how this turned out.